packages/shared/lib/services/sync/orchestrator.service.ts

```diff
@@ -404,6 +358,59 @@ export class Orchestrator {
             );
         }
     }
+    private async fetchSyncData(syncId: string, environmentId: number) {
+        const syncClient = await SyncClient.getInstance();
+        const schedule = await getSchedule(syncId);
+        const latestJob = await getLatestSyncJob(syncId);
+        let status = this.classifySyncStatus(latestJob?.status as SyncStatus, schedule?.status as ScheduleStatus);
+
+        const syncSchedule = await syncClient?.describeSchedule(schedule?.schedule_id as string);
+        if (syncSchedule) {
+            if (syncSchedule?.schedule?.state?.paused && status !== SyncStatus.PAUSED) {
+                await updateScheduleStatus(schedule?.id as string, SyncCommand.PAUSE, null, environmentId);
+                status = SyncStatus.PAUSED;
+            } else if (!syncSchedule?.schedule?.state?.paused && status === SyncStatus.PAUSED) {
+                await updateScheduleStatus(schedule?.id as string, SyncCommand.UNPAUSE, null, environmentId);
+                status = SyncStatus.STOPPED;
+            }
+        }
+
+        let nextScheduledSyncAt = null;
+        if (status !== SyncStatus.PAUSED) {
+            if (syncSchedule && syncSchedule?.info && syncSchedule?.info.futureActionTimes && syncSchedule?.info?.futureActionTimes?.length > 0) {
+                const futureRun = syncSchedule.info.futureActionTimes[0];
+                nextScheduledSyncAt = syncClient?.formatFutureRun(futureRun?.seconds?.toNumber() as number);
+            }
+        }
+
+        return { schedule, latestJob, status, nextScheduledSyncAt };
+    }
+
+    private async buildReportedStatus(
```

Comment:
what about simply `reportedStatus()` as the function name

Reply:
Updated with https://github.com/NangoHQ/nango/pull/1576/commits/286f9a529cd2c4bcc05d345db1fd24972a9ae5db